### PR TITLE
Made some Appender initialization lazy, so that slf4j doesn't need to…

### LINKED ISSUE #354: SentryAppender logs messages when JNDI isn't configured
The exception handling at https://github.com/getsentry/raven-java/blob/master/raven/src/main/java/com/getsentry/raven/config/JndiLookup.java#L37 logs messages; unfortunately, this is called by **SentryAppender** during logback/slf4j startup, resulting in this nastygram written by logback:

```
SLF4J: A number (6) of logging calls during the initialization phase have been intercepted and are
SLF4J: now being replayed. These are subject to the filtering rules of the underlying logging system.
SLF4J: See also http://www.slf4j.org/codes.html#replay
```

This makes a mess for any program trying to handle the stdout/stderr from the JVM in a reasonable way.

### PR DESCRIPTION
**Pull request by bretthoerner**
… replay logs and system property or environment variable configuration override hardcoded file configuration.

This fixes #354 and #266

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -1,7 +1,8 @@
 Version 7.8.7
 -------------
 
--
+- Made some Appender initialization lazy, so that slf4j doesn't need to replay logs and
+  system property or environment variable configuration override hardcoded file configuration.
```
Comment (tkaemming) [inline CHANGES]: This is a strange sentence. Maybe reword?

File: raven-log4j/src/main/java/com/getsentry/raven/log4j/SentryAppender.java [hunk 2]
```diff
@@ -110,6 +107,54 @@ public SentryAppender(Raven raven) {
     }
 
     /**
+     * Do some appender initialization *after* instance construction, so that we don't
+     * log in the constructor (which can cause annoying messages) and so that system
+     * properties and environment variables override hardcoded appender configuration.
+     */
+    @SuppressWarnings("checkstyle:hiddenfield")
+    private void lazyInit() {
```
Comment (tkaemming) [inline raven-log4j/src/main/java/com/getsentry/raven/log4j/SentryAppender.java]: Why isn't this just part of or called from `initRaven`? (I didn't look too closely.)

Comment (bretthoerner) [inline raven-log4j/src/main/java/com/getsentry/raven/log4j/SentryAppender.java]: One of the constructors takes `Raven` as a parameter, so `raven` being not-null isn't enough of a check for this.

Comment (tkaemming) [inline raven-log4j/src/main/java/com/getsentry/raven/log4j/SentryAppender.java]: OK, I actually read the code this time. 😬 

I think what I was noting was mostly that this pattern seems to exist side-by-side in each usage:

```java
lazyInit();
if (raven == null) {
    initRaven();
}
```

It makes sense that is initialization check can't exist within the `if` block, but can't we combine these two methods somehow into a standard pattern for initializing settings if necessary, then instantiating the `Raven` instance if necessary, rather than having them independent like this? Or was there a reason the `initRaven` method had to be wrapped in the conditional (rather than containing the conditional) that I'm forgetting?

**Comment by bretthoerner** [comment]
Ohhhh, no you're right the Raven init could move inside.

File: CHANGES (hunk 1 above)
Comment (bretthoerner) [inline CHANGES]: Good call, I broke it out into 2 more user-facing notes.

File: raven-log4j/src/main/java/com/getsentry/raven/log4j/SentryAppender.java (hunk 2 above)
Comment (bretthoerner) [inline raven-log4j/src/main/java/com/getsentry/raven/log4j/SentryAppender.java]: Oh, yeah, I moved the Raven check into `lazyInit` (as it is lazy initialization too).
